djangocms_pageadmin/admin.py

```diff
@@ -92,6 +92,28 @@ def get_queryset(self, request):
             )
         )
 
+    def get_search_results(self, request, queryset, search_term):
+        """
+        Override the ModelAdmin method for fetching search results to filter for urls associated with the pagecontent
+        :param request: PageContent Admin request
+        :param queryset: PageContent queryset
+        :param search_term: Term to be searched for
+        :return: results
+        """
+        language = get_language()
+        queryset, use_distinct = super().get_search_results(request, queryset, search_term)
+        original_queryset = queryset
+        # Filter by language to avoid hits from URLs in different languages
+        queryset |= queryset.filter(
+            Q(page__urls__slug__icontains=search_term) | Q(page__urls__path__icontains=search_term),
+            page__urls__language=language
+        )
+        # This is a workaround to avoid bringing in much of the code which decides whether a queryset is distinct.
+        # https://github.com/django/django/blob/2a62cdcfec85938f40abb2e9e6a9ff497e02afe8/django/contrib/admin/options.py#L980 # NOQA
+        if not list(queryset) == list(original_queryset):
```

Comment:
To avoid this, would filtering the queryset initially and passing the result to the super call work? 
```
    def get_search_results(self, request, queryset, search_term):
        """
        Override the ModelAdmin method for fetching search results to filter for urls associated with the pagecontent
        :param request: PageContent Admin request
        :param queryset: PageContent queryset
        :param search_term: Term to be searched for
        :return: results
        """
        language = get_language()
        queryset |= queryset.filter(
            Q(page__urls__slug__icontains=search_term) | Q(page__urls__path__icontains=search_term),
            page__urls__language=language
        )
        return super().get_search_results(request, queryset, search_term)
```

Reply:
I've updated this so that it just uses distinct if there is a search_term, i.e. whenever the search is not blank. This is because the super is going off of only the title as a search field, and therefore it won't be consistent with the custom logic for URLs. There is precedent for this approach in the internal URL manager, so I think it is the best choice. 

It would be useful if you could have another look at this method, I've had to rework how we handle the querysets to make the search work with symbols within our URLS.

Reply:
Right think I understand what is going on here more now, I was wondering why we cant set the `page__urls__slug` and `page__urls__path` directly on the `search_fields` attribute, but that will cause results to appear for the incorrect language. 

And the logic around `use_distinct` is making more sense to me now, as I looked in to the `lookup_needs_distinct` [util](https://github.com/django/django/blob/stable/3.2.x/django/contrib/admin/utils.py#L28) that gets used by the django internal `get_search_results` method, which checks if any of the search fields are a M2M relation. Because the `search_fields` attribute is set to just `title`, the super call in django will return `False`. But as we are using the addition custom filter across relations we know that  `use_distinct` will need to be True.

So my only suggestion really is to improve your comment about it to explain why we hardcode it to True because of the above.

Reply:
Have added a better comment!